packages/cli-kit/src/git.ts

```diff
@@ -30,12 +31,38 @@ export const DetachedHeadError = () => {
   )
 }
 
+export const NothingToCommitError = () => {
+  return new Abort('Nothing to commit')
+}
+
 export async function initializeRepository(directory: string) {
   debug(content`Initializing git repository at ${token.path(directory)}...`)
   await ensurePresentOrAbort()
   await git(directory).init()
 }
 
+export async function createGitIngore(directory: string) {
```

Comment:
I'm not sure this is the appropriate place for this function in this package since it's specific to Hydrogen. Could this be a part of `create-hydrogen` instead? Or this function could be more generic where it takes in both a directory and content but by that point it wouldn't be doing much.